utils/azureCredentialsManager.ts

```diff
@@ -135,4 +135,292 @@ export class AzureCredentialsManager {
+                } else {
+                    return [];
+                }
+            });
+
+            await request.get('https://' + element.registry.loginServer + '/v2/' + element.name + '/tags/list', {
+                auth: {
+                    bearer: accessTokenARC
+                }
+            }, (err, httpResponse, body) => {
+                if (err) { return []; }
+
+                if (body.length > 0) {
+                    tags = JSON.parse(body).tags;
+                }
+            });
+
+            for (let tag of tags) {
+                image = new AzureImage(azureAccount, element.registry, element, tag, element.subscription, element.resourceGroupName, accessTokenARC, refreshTokenARC, element.password, element.username);
+                allImages.push(image);
+            }
+        }
+        return allImages;
+    }
+
+    //Implements new Service principal model for ACR container registries while maintaining old admin enabled use
+    /**
+     * this function implements a new Service principal model for ACR and gets the valid login credentials to make an API call
+     * @param subscription : the subscription the registry is on
+     * @param registry : the registry to get login credentials for
+     * @param context : if command is invoked through a right click on an AzureRepositoryNode. This context has a password and username
+     */
+    public async loginCredentials(subscription: SubscriptionModels.Subscription, registry: Registry, context?: AzureImageNode | AzureRepositoryNode): Promise<{ password: string, username: string }> {
+        let node: AzureImageNode | AzureRepositoryNode;
+        if (context) {
+            node = context;
+        }
+        let username: string;
+        let password: string;
+        const client = this.getContainerRegistryManagementClient(subscription);
+        const resourceGroup: string = registry.id.slice(registry.id.search('resourceGroups/') + 'resourceGroups/'.length, registry.id.search('/providers/'));
+        if (context) {
+            username = node.userName;
+            password = node.password;
+        } else if (registry.adminUserEnabled) {
+            let creds = await client.registries.listCredentials(resourceGroup, registry.name);
+            password = creds.passwords[0].value;
+            username = creds.username;
+        } else {
+            //grab the access token to be used as a password, and a generic username
+            let creds = await this.getTokens(registry);
+            password = creds.accessToken;
+            username = '00000000-0000-0000-0000-000000000000';
+        }
+        return { password, username };
+    }
+
+    /**
+     * function to allow user to pick a desired image for use
+     * @param repository the repository to look in
+     * @returns an AzureImage object (see azureUtils.ts)
+     */
+    public async getImage(repository: Repository): Promise<AzureImage> {
```

Comment:
Similar to those below these are helpful commands that can be reused but don't fit the model of this class, alternatively names like quickPickACRImage would make the functionality clearer.